### PR TITLE
Make sure Root CAs can be used for backend calls when TLS isn't used

### PR DESCRIPTION
**Pull request by leitzler**
## Description
This PR makes sure that the Root CAs feature can be used even when TLS isn't configured.

## Motivation and Context
TLS configuration and Root CAs configuration are two separate things. Currently they are both read by `getSSLConfig(...)` and if TLS isn't configured, Root CAs won't be read either.

## How Has This Been Tested?
`make test` and manually by adding a custom root certificate for a backend without configure minio for TLS.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by nitisht** [comment]
@leitzler can you PTAL at the Travis failure

**Comment by leitzler** [comment]
Strange, looks like a flaky test. My change shouldn't alter the behavior (much less on Windows alone).
It looks like this test case have failed before, e.g.: https://travis-ci.org/minio/minio/jobs/444918409 when only the documentation changed.

I'll retrigger the tests. Might be something to open a new issue for tough.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by harshavardhana** [comment]
What is the purpose of just having root CAs v/s TLS ? We don't have any other good reason to add new CAs other than trusting each other peers. 

Minio is not like mc which already does this type of behavior .

**Comment by leitzler** [comment]
I haven't looked into details of all the places it is used, just noticed that it was in several files.
It came from the [documentation](https://docs.minio.io/docs/how-to-secure-access-to-minio-server-with-tls.html) that says:
> Minio can be configured to connect to other servers, whether Minio nodes or servers like NATs, Redis. If these servers use certificates that are not registered in one of the known certificates authorities, you can make Minio server trust these CAs by dropping these certificates under Minio config path (~/.minio/certs/CAs/ on Linux or C:\Users\<Username>\.minio\certs\CAs on Windows).

**Comment by harshavardhana** [comment]
> I haven't looked into details of all the places it is used, just noticed that it was in several files.
> It came from the [documentation](https://docs.minio.io/docs/how-to-secure-access-to-minio-server-with-tls.html) that says:

Yes, it is written along in the TLS documentation, RootCAs are not useful for Minio other than TLS configuration. That is why they are together.  Minio doesn't talk to some other external entity which requires additional CA trusting - its not a general purpose command line tool like `mc`. 

So I don't think this PR is needed at the moment @leitzler

**Comment by leitzler** [comment]
I see, from the documentation I assumed that you could use minio in gateway mode with a backend that require a custom CA. That doesn't require the connection towards minio to use tls.

Just to make sure, as an example. I'm not supposed to set `MINIO_LOGGER_HTTP_ENDPOINT` to a endpoint that require a root CA without also configure minio to use tls?

No worries if you don't want it merged, I didn't write an issue before submitting it so I know that the PR might not get accepted :)

**Comment by harshavardhana** [comment]
> I see, from the documentation I assumed that you could use minio in gateway mode with a backend that require a custom CA. That doesn't require the connection towards minio to use tls.
> 
> Just to make sure, as an example. I'm not supposed to set `MINIO_LOGGER_HTTP_ENDPOINT` to a endpoint that require a root CA without also configure minio to use tls?
> 
> No worries if you don't want it merged, I didn't write an issue before submitting it so I know that the PR might not get accepted :)

Oh yes, you are right good point - I had missed that @leitzler - then this PR makes sense.

File: cmd/certs.go [hunk 1]
```diff
@@ -150,24 +150,24 @@ func loadX509KeyPair(certFile, keyFile string) (tls.Certificate, error) {
...
 	c, err = certs.New(getPublicCertFile(), getPrivateKeyFile(), loadX509KeyPair)
 	if err != nil {
-		return nil, nil, nil, false, err
+		return nil, nil, false, err
 	}
 
-	if rootCAs, err = getRootCAs(getCADir()); err != nil {
-		return nil, nil, nil, false, err
-	}
+	//if rootCAs, err = getRootCAs(getCADir()); err != nil {
```
Comment (harshavardhana) [inline cmd/certs.go]: Don't need to comment like this @leitzler can you remove this?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/certs.go (hunk 1 above)
Comment (leitzler) [reply cmd/certs.go]: Oh, good catch, sorry about that

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Review by nitisht — APPROVED** [review]

**Pull request merged**